server/src/index.js

```diff
@@ -0,0 +1,223 @@
+
+import express from 'express';
+import cors from 'cors';
+import { EventEmitter } from 'events';
+import { WebSocketServer } from 'ws';
+import { WebSocket } from 'ws';
+import { v4 as uuidv4 } from 'uuid';
+import { connectDB, syncDB } from './utils/db.js';
+import { Room } from './models/room.js';
+import { Message } from './models/message.js';
+import { error } from 'console';
+
+const app = express();
+
+connectDB();
+syncDB();
+
+app.use(cors());
+app.use(express.json());
+
+const emitter = new EventEmitter();
+const rooms = new Map();
+const messages = [];
+
+let connectionsCount = 0;
+
+app.post('/rooms', async (req, res) => {
+  const { roomName } = req.body;
+  try {
+    const existingRoom = await Room.findOne({ where: { roomName } });
+    if (existingRoom) {
+      return res.status(409).json({ error: 'Room already exists' });
+    }
+    const newRoom = await Room.create({ roomName });
+    res.status(201).json({ roomId: newRoom.id, roomName: newRoom.roomName });
+  } catch (error) {
+    console.error('Error creating room:', error);
+    res.status(500).json({ error: 'Internal Server Error' });
+  }
+});
+
+app.get('/rooms', async (req, res) => {
+  try {
+    const rooms = await Room.findAll({ attributes: ['id', 'roomName'] });
+    res.status(200).json(rooms.map(room => ({ roomId: room.id, roomName: room.roomName })))
+  } catch (error) {
+    console.error('Error fetching rooms:', error);
+    res.status(500).json({ error: 'Internal Server Error' });
+  }
+});
+
+app.post('/messages', async (req, res) => {
+  const { text, roomId, username } = req.body;
+
+  try {
+    const room = await Room.findByPk(roomId);
+    if (!room) {
+      return res.status(404).json({ error: 'Room not found' });
+    }
+
+    const message = await Message.create({
+      text,
+      username,
+      RoomId: roomId,
+    });
+
+    res.status(201).json({ success: true, message });
+  } catch (error) {
+    console.error('Error adding message:', error);
+    res.status(500).json({ error: 'Internal Server Error' });
+  }
+});
+
+app.get('/connections-count', (req, res) => {
+  res.status(200).json({ connectionsCount });
+});
+
+app.get('/messages', (req, res) => {
+  res.status(405).send("Method Not Allowed");
+});
+
+app.get('/rooms/:roomId/messages', async (req, res) => {
+  const { roomId } = req.params;
+
+  try {
+    const messages = await Message.findAll({
+      where: { RoomId: roomId },
+      order: [['createdAt', 'ASC']],
+    });
+
+    res.json(messages);
+  } catch (error) {
+    console.error("Error loading messages:", error);
+    res.status(500).send("Error loading messages.");
+  }
+});
+
+const PORT = 3005;
+const server = app.listen(PORT, () => {
+  console.log(`Server is running on http://localhost:${PORT}`);
+});
+
+const wss = new WebSocketServer({ server });
+
+wss.on('connection', (connection) => {
+  connectionsCount++;
+  console.log('connection');
+  console.log(connectionsCount);
+
+  connection.on('open', async () => {
```

Comment:
The 'open' event for WebSocket connections is not a standard event. Consider using the 'connection' event to handle initial setup tasks for WebSocket connections.